src/processors.zig

```diff
@@ -181,6 +179,46 @@ pub const CrossPageLinkProcessor = struct {
             }
         }
     }
+    fn parseAltText(
+        pctx: anytype,
+        alt_text: []const u8,
+        maybe_scale: ?[]const u8,
+    ) !void {
+        if (maybe_scale) |scale| {
+            if (parseResolutionFromAlt(scale)) |res| {
+                try pctx.out.print("alt=\"{s}\"", .{alt_text});
+                try pctx.out.print("width=\"{s}\"", .{res[0].?});
+                if (res[1] != null) try pctx.out.print("height=\"{s}\"", .{res[1].?});
+                return;
+            }
+
+            // Third position exists but is not a resolution
+            // Treat scale as a literal value and reconstruct what would have been the alt text
+            try pctx.out.print("alt=\"{s}|{s}\"", .{ alt_text, scale });
+            return;
+        }
+
+        if (parseResolutionFromAlt(alt_text)) |res| {
+            try pctx.out.print("width=\"{s}\"", .{res[0].?});
+            if (res[1] != null) try pctx.out.print("height=\"{s}\"", .{res[1].?});
+            return;
+        }
+
+        try pctx.out.print("alt=\"{s}\"", .{alt_text});
+    }
+
+    fn parseResolutionFromAlt(alt: []const u8) ?[2]?[]const u8 {
+        _ = std.fmt.parseInt(i32, alt, 10) catch {
+            var resolutionSplit = std.mem.split(u8, alt, "x");
+            const width = resolutionSplit.next();
+            const height = resolutionSplit.next();
+
+            _ = std.fmt.parseInt(i32, width.?, 10) catch return null;
```

Comment:
if `width` is null, shouldn't we ignore it?

(note, `.?` is an assertion that something is not null, it is runtime-checked safety, which panics on debug/releasesafe, undefined behavior on releasefast/releasesmall)

Reply:
Ah yeah, this should've returned early if `width` is null. We need the `.?` so that we pass the typecheck for parseInt.

Reply:
width is still nullable, you can do `width orelse return null` so that its ignored and typechecks

Reply:
oh that's a lot better